### PR TITLE
Adding face

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This PR adds the shape.wire attribute so that a stp file with the original wire made from points can be exported

This wire is just the points and is made prior to any CAD extrusion or boolean operations.

So if the component has cuts or unions or intersections then this wire doesn't fully represent the component 2D profile.

For shapes like the plasma which don't normally undergo additional CAD operations the 2D wire is a nice feature to have.

Additionally this potentially opens up the door to using offset2D to make wires from wires with a nice offset which could be useful for nested geometry.

```python
import paramak
paramak.Plasma().export_stp(solid_or_wire='wire')
```
Screenshot from 2020-12-09 13-08-33

reactors and components can also be exported using the same ```.export_stp(solid_or_wire='wire')``` method

However due to the use of CAD operations in those components the wires don't look great.

Screenshot from 2020-12-09 13-09-15

Still I think this is a worth while addition even if it is not useful everywhere.

One thing I would be really keen on changing in this PR is reducing the amount of duplication in the new tests. If anyone has any suggestions there that would be great. I have basically used the same function several times over and was hoping that there is a better way.

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Shimwell** [comment]
I am keen to extend this to stl files but am the export for a wire doesn't appear to produce anything in the output stl file made

https://github.com/CadQuery/cadquery/issues/537

File: paramak/reactor.py [hunk 1]
```diff
@@ -270,7 +270,8 @@ def export_neutronics_description(
 
         return str(path_filename)
 
-    def export_stp(self, output_folder="", graveyard_offset=100):
+    def export_stp(self, output_folder="", graveyard_offset=100,
+                   solid_or_wire='solid'):
```
Comment (RemDelaporteMathurin) [inline paramak/reactor.py:274]:
```suggestion
                   mode='solid'):
```

I think solid_or_wire is a bit heavy. Could something like "mode" do the trick ?

File: paramak/shape.py [hunk 2]
```diff
@@ -739,7 +756,7 @@ def export_stl(self, filename, tolerance=0.001):
 
         return str(path_filename)
 
-    def export_stp(self, filename=None, units='mm'):
+    def export_stp(self, filename=None, units='mm', solid_or_wire='solid'):
```
Comment (RemDelaporteMathurin) [inline paramak/shape.py:759]:
```suggestion
    def export_stp(self, filename=None, units='mm', mode='solid'):
```

Same as above, could solid_or_wire be replaced by mode ?

File: paramak/shape.py [hunk 3]
```diff
@@ -764,7 +784,13 @@ def export_stp(self, filename=None, units='mm'):
             path_filename = Path(self.stp_filename)
 
         with open(path_filename, "w") as out_file:
-            exporters.exportShape(self.solid, "STEP", out_file)
+            if solid_or_wire == 'solid':
+                exporters.exportShape(self.solid, "STEP", out_file)
+            elif solid_or_wire == 'wire':
+                exporters.exportShape(self.wire, "STEP", out_file)
+            else:
+                raise ValueError("The solid_or_wire argument for export_stp \
```
Comment (RemDelaporteMathurin) [inline paramak/shape.py:792]: This line isn't tested.

File: tests/test_parametric_shapes/test_RotateMixedShape.py [hunk 4]
```diff
@@ -181,6 +181,31 @@ def test_export_stl(self):
         assert Path("filename.stl").exists() is True
         os.system("rm filename.stl")
 
+    def test_export_stp(self):
```
Comment (RemDelaporteMathurin) [inline tests/test_parametric_shapes/test_RotateMixedShape.py:184]: There is another test_export_stp in this file.

File: tests/test_parametric_shapes/test_RotateStraightShape.py [hunk 5]
```diff
@@ -7,6 +7,9 @@
 import pytest
 from paramak import RotateStraightShape
 
+import os
+from pathlib import Path
```
Comment (RemDelaporteMathurin) [inline tests/test_parametric_shapes/test_RotateStraightShape.py:11]: Code inspector suggests to move imports arround.

File: tests/test_parametric_shapes/test_RotateStraightShape.py [hunk 6]
```diff
@@ -222,6 +225,31 @@ def test_hash_value(self):
         assert self.test_shape.solid is not None
         assert initial_hash_value != self.test_shape.hash_value
 
+    def test_export_stp(self):
```
Comment (RemDelaporteMathurin) [inline tests/test_parametric_shapes/test_RotateStraightShape.py:228]: There is another test_export_stp in this file.

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Hi @Shimwell , just a few tiny comments. I don't know what to do about code duplication here.

File: tests/test_parametric_shapes/test_ExtrudeMixedShape.py [hunk 7]
```diff
@@ -130,6 +130,31 @@ def test_extrude_both(self):
         assert self.test_shape.volume == pytest.approx(
             test_volume_extrude_both)
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeMixedShape.py:139]: Minor refactoring
os.system("rm test_solid.stp test_solid2.stp test_wire.stp")

File: tests/test_parametric_components/test_TestTFCoilCasing.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        assert self.test_shape.solid is not None
+
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_components/test_TestTFCoilCasing.py:39]: Minor refactoring
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeCircleShape.py [hunk 9]
```diff
@@ -93,6 +94,31 @@ def test_extrude_both(self):
         assert self.test_shape.volume == pytest.approx(
             test_volume_extrude_both)
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeCircleShape.py:103]: Minor refactoring
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeMixedShape.py [hunk 10]
```diff
@@ -130,6 +130,31 @@ def test_extrude_both(self):
         assert self.test_shape.volume == pytest.approx(
             test_volume_extrude_both)
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeMixedShape.py:137]: As above

File: tests/test_parametric_components/test_TestTFCoilCasing.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_components/test_TestTFCoilCasing.py:54]: Same minor refactoring
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeCircleShape.py [hunk 12]
```diff
@@ -93,6 +94,29 @@ def test_extrude_both(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeCircleShape.py:118]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeMixedShape.py (hunk 7 above)
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeMixedShape.py:139]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeMixedShape.py [hunk 13]
```diff
@@ -130,6 +130,31 @@ def test_extrude_both(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeMixedShape.py:156]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeSplineShape.py [hunk 14]
```diff
@@ -84,6 +86,31 @@ def test_extrude_both(self):
         assert self.test_shape.volume == pytest.approx(
             test_volume_extrude_both)
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeSplineShape.py:95]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeSplineShape.py [hunk 15]
```diff
@@ -84,6 +86,31 @@ def test_extrude_both(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeSplineShape.py:112]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeStraightShape.py [hunk 16]
```diff
@@ -98,6 +100,31 @@ def test_extrude_both(self):
         assert self.test_shape.volume == pytest.approx(
             test_volume_extrude_both)
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeStraightShape.py:109]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeStraightShape.py [hunk 17]
```diff
@@ -98,6 +100,31 @@ def test_extrude_both(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeStraightShape.py:126]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateCircleShape.py [hunk 18]
```diff
@@ -81,6 +83,32 @@ def test_cut_volume(self):
             outer_shape_volume - self.test_shape.volume
         )
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateCircleShape.py:92]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateCircleShape.py [hunk 19]
```diff
@@ -81,6 +83,32 @@ def test_cut_volume(self):
...
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        # the circle wire file is actually larger than the circle solid file
+        # assert Path("test_wire.stp").stat().st_size < \
+        #     Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateCircleShape.py:110]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateMixedShape.py [hunk 20]
```diff
@@ -181,6 +181,31 @@ def test_export_stl(self):
...
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateMixedShape.py:190]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateMixedShape.py [hunk 21]
```diff
@@ -181,6 +181,31 @@ def test_export_stl(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateMixedShape.py:207]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateSplineShape.py [hunk 22]
```diff
@@ -67,6 +69,31 @@ def test_shape_face_areas(self):
         assert len(self.test_shape.areas) == 3
         assert len(set([round(i) for i in self.test_shape.areas])) == 2
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateSplineShape.py:78]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateSplineShape.py [hunk 23]
```diff
@@ -67,6 +69,31 @@ def test_shape_face_areas(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateSplineShape.py:95]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateStraightShape.py [hunk 24]
```diff
@@ -222,6 +225,31 @@ def test_hash_value(self):
...
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateStraightShape.py:234]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateStraightShape.py [hunk 25]
```diff
@@ -222,6 +225,31 @@ def test_hash_value(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_RotateStraightShape.py:251]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepCircleShape.py [hunk 26]
```diff
@@ -130,6 +132,31 @@ def test_surface_count(self):
         assert len(self.test_shape.areas) == 3
         assert len(set(round(i) for i in self.test_shape.areas)) == 2
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepCircleShape.py:141]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepCircleShape.py [hunk 27]
```diff
@@ -130,6 +132,31 @@ def test_surface_count(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepCircleShape.py:158]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepMixedShape.py [hunk 28]
```diff
@@ -123,6 +125,31 @@ def test_surface_count(self):
         assert len(self.test_shape.areas) == 6
         assert len(set(round(i) for i in self.test_shape.areas)) == 5
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepMixedShape.py:134]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepMixedShape.py [hunk 29]
```diff
@@ -123,6 +125,31 @@ def test_surface_count(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepMixedShape.py:151]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepSplineShape.py [hunk 30]
```diff
@@ -94,6 +96,31 @@ def test_surface_count(self):
         assert len(self.test_shape.areas) == 3
         assert len(set(round(i) for i in self.test_shape.areas)) == 2
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepSplineShape.py:105]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepSplineShape.py [hunk 31]
```diff
@@ -94,6 +96,31 @@ def test_surface_count(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepSplineShape.py:122]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepStraightShape.py [hunk 32]
```diff
@@ -82,6 +84,56 @@ def test_surface_count(self):
         assert len(self.test_shape.areas) == 6
         assert len(set(round(i) for i in self.test_shape.areas)) == 3
 
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepStraightShape.py:93]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepStraightShape.py [hunk 33]
```diff
@@ -82,6 +84,56 @@ def test_surface_count(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepStraightShape.py:110]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepStraightShape.py [hunk 34]
```diff
@@ -82,6 +84,56 @@ def test_surface_count(self):
...
+
+    def test_export_stp(self):
+        """Exports and stp file with solid_or_wire = solid and wire and checks
+        that the outputs exist and relative file sizes are correct."""
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepStraightShape.py:118]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_SweepStraightShape.py [hunk 35]
```diff
@@ -82,6 +84,56 @@ def test_surface_count(self):
...
+        assert Path("test_wire.stp").exists() is True
+
+        assert Path("test_solid.stp").stat().st_size == \
+            Path("test_solid2.stp").stat().st_size
+        assert Path("test_wire.stp").stat().st_size < \
+            Path("test_solid2.stp").stat().st_size
+
+        os.system("rm test_solid.stp")
+        os.system("rm test_solid2.stp")
+        os.system("rm test_wire.stp")
```
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_SweepStraightShape.py:135]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_ExtrudeCircleShape.py (hunk 12 above)
Comment (billingsley-john) [inline tests/test_parametric_shapes/test_ExtrudeCircleShape.py:118]: As above
```suggestion
        os.system("rm test_solid.stp test_solid2.stp test_wire.stp")
```

File: tests/test_parametric_shapes/test_RotateStraightShape.py (hunk 6 above)
Comment (Shimwell) [reply tests/test_parametric_shapes/test_RotateStraightShape.py:225]: renamed test_export_stp method, thanks for spotting that

File: paramak/shape.py (hunk 3 above)
Comment (Shimwell) [reply paramak/shape.py]: added a test

File: tests/test_parametric_shapes/test_RotateStraightShape.py (hunk 5 above)
Comment (Shimwell) [reply tests/test_parametric_shapes/test_RotateStraightShape.py]: now sorted imports

File: tests/test_parametric_shapes/test_RotateMixedShape.py (hunk 4 above)
Comment (Shimwell) [reply tests/test_parametric_shapes/test_RotateMixedShape.py:184]: renamed the method

File: tests/test_parametric_shapes/test_ExtrudeCircleShape.py (hunk 12 above)
Comment (Shimwell) [reply tests/test_parametric_shapes/test_ExtrudeCircleShape.py]: committed

**Review by RemDelaporteMathurin — APPROVED** [review]

**Pull request merged**